boilerplate/app/navigators/navigationUtilities.ts

```diff
@@ -147,8 +147,11 @@ export function useNavigationPersistence(storage: Storage, persistenceKey: strin
 
   const restoreState = async () => {
     try {
+      const initialUrl = await Linking.getInitialURL()
       const state = (await storage.load(persistenceKey)) as NavigationProps["initialState"] | null
-      if (state) setInitialNavigationState(state)
+
+      // Only restore the state if app has not started from a deep link
+      if (state && !initialUrl) setInitialNavigationState(state)
```

Comment:
What about checking if `initialUrl` exists first and if it doesn't _then_ do the state lookup in async storage?

This way on a deep link we skip that load all together?

Reply:
That's a really good idea, I'll update the code
If the solution works, I will also submit a PR in react-navigation documents

Reply:
I seemed to have success with

```
      const initialUrl = await Linking.getInitialURL()

      if (!initialUrl) {
        const state = (await storage.load(persistenceKey)) as NavigationProps["initialState"] | null
        if (state) setInitialNavigationState(state)
      }
```

But still double check me and then we can get this in

Reply:
Fortunately i have my production app open in front of me, give me a min

Reply:
@frankcalise Tested and Done